src/notebook_utils/automation.py

```diff
@@ -14,6 +14,67 @@
 def is_ipynb_checkpoint(notebook_file: PathLike):
     return ".ipynb_checkpoint" in str(notebook_file)
 
+def run_notebook(notebook_path: PathLike):
+    return subprocess.run(["jupyter", "nbconvert", "--to", "notebook", "--inplace", "--execute", "--ExecutePreprocessor.timeout=600", f"{notebook_path}"], capture_output=True, text=True)
+
+def run_notebook_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-f",
+        "--file",
+        type=pathlib.Path,
+        help="Notebook to run."
+    )
+
+    parser.add_argument("-v", "--verbose", action="store_true", help="Output logging details.")
+    args = parser.parse_args()
+
+    if args.verbose:
+        _logger.setLevel(logging.INFO)
+
+    if args.file is None or args.file.suffix != ".ipynb":
+        _logger.error("You must specify a valid notebook (.ipynb file) to run.")
+        sys.exit(1)
```

Comment:
You can define an action (inherit from `argparse.Action`, and pass it to with `action=Class` to `add_argument`) to perform all these kind of checks during `__call__`. If invalid, raise an argument error directly. 
See example: https://github.com/crim-ca/weaver/blob/e39ffccf3643a70b5496ca8b8799fa1a903c1a66/weaver/cli.py#L1490-L1499